### ISSUE TITLE
Filename for fastsimcoal

### ISSUE THREAD

**Issue by Rilquer** [isaacovercast/easySFS#9]
Hi, Isaac.

Thanks for this script, it was really useful. I came up with a minor issue, I'm not sure if it's a big problem. The output file for the joint MAF is named '_jointMAFpop0_1.obs', but fastsimcoal requires 'jointMAFpop1_0.obs', when there are two populations (which is my case). So I renamed the file and created a tranposed matrix (with col names being d1 and row names being d0) based on the one the script outputs (which is the inverse, col names being d0 and row names being d1). I was wondering if that is okay, as I'm not very familiar with the logic behind the files describing the different types of SFSs.

Thank you in advance.

**Comment by henryarenasc** [comment]
Hi Isaac,
I had the same issue. I'm concerned whether it is only an issue about the naming of the file or beside renaming the file I also should transpose the matrix.
Cheers, Henry

**Comment by Rilquer** [comment]
Hi, Henry.
I recently emailed Isaac about this issue. Here is how it went:

I wrote:
"Hi, Isaac.

I was working with script you created for easySFS, and came up with a minor question, I'm not sure if it's a big problem. The output file for the joint MAF is named '_jointMAFpop0_1.obs', but fastsimcoal requires 'jointMAFpop1_0.obs', when there are two populations (which is my case). So I renamed the file and created a tranposed matrix (with col names being d1 and row names being d0) based on the one the script outputs (which is the inverse, col names being d0 and row names being d1). I was wondering if that is okay, as I'm not very familiar with the logic behind the files describing the different types of SFSs."

Isaac replied:
"Hi Rilquer,
Yeah, that sounds fine to me. fsc is a little picky with formatting sometimes, sorry my script is buggy, I'll fix it."

Hope it helps.

**Comment by ab08028** [comment]
I'm also in a bit of a conundrum about this. When I have 2D SFSes with different sample sizes, I find that the population IDs are switched between the filename and the rows/columns, for example:

pop0 has projection size 12
pop1 has projection size 14

But my resulting file pop0_1.obs has d0_0 - d0_14 along the top
and d1_0 - d1_12 along the rows. So you could just switch the labels of d0 and d1 and it would reference the appropriate population, but you may also need to transpose the matrix so that the d1s are down the side rather than along the top.

 Could lines 116-123 be the relevant lines of code where the switch is happening? It looks to me as though the popidx[pair[]] may be switched between the column names and row_headers (** marks values I think should perhaps be switched)

```
        with open(fsc_twoD_filename, 'w') as outfile:
            outfile.write("1 observation\n")
            ## Format column headers (i.e. d0_0 d0_1 d0_2 .. d0_n for deme 0 up to sample size of n)
            outfile.write("\t" + "\t".join(["d{}_".format(popidx[pair[**0**]]) + str(x) for x in xrange(projPairs[i][**1**]+1)]) + "\n") 

            ## Format row headers
            row_headers = ["d{}_".format(popidx[pair[**1**]]) + str(x) for x in xrange(projPairs[i][**0**]+1)]
```


Should it be? --> 
```
        with open(fsc_twoD_filename, 'w') as outfile:
            outfile.write("1 observation\n")
            ## Format column headers (i.e. d0_0 d0_1 d0_2 .. d0_n for deme 0 up to sample size of n)
            outfile.write("\t" + "\t".join(["d{}_".format(popidx[pair[**1**]]) + str(x) for x in xrange(projPairs[i][1]+1)]) + "\n") 

            ## Format row headers
            row_headers = ["d{}_".format(popidx[pair[**0**]]) + str(x) for x in xrange(projPairs[i][0]+1)]
```
 Then at least the population IDs are consistent between the filename and the row/column names. However, I'm not sure if you then need to additionally transpose the matrix for fsc to take it as input?

What do you you all think? 

Cheers,
~ Annabel

**Comment by isaacovercast** [comment]
Hi Annabel,
Yes, this is right, the id in the filename is transposed. I made the change you suggested so now the ID and the row/col of each jSFS agree. The only issue now is that the file names are *pop0_1.obs, whereas the fsc2 manual says this should be *pop1_0.obs. It _SHOULD_ work since the first id indicates the data in the row and the second indicates the data in the column, and this is how it is now in the easySFS output files. If you can test this and let me know if it works okay that would be wonderful.

**Comment by ab08028** [comment]
Thanks, Isaac! When we were running it we found that fsc2 wasn't liking the "d2, d3, ..." labeling when we were running 2D models so we ended up having to rename each file to be pop1_0 regardless of what the population numbers were from easySFS, with the column/row names as d0/d1, respectively. I haven't tried with >2D models, so I'm not sure how things might sort out then. But at least for now it does seem like other users are right that the filename does need to be 1_0 with 1 corresponding to the population that is down the rows, and 0 the population along the columns. Fsc is so finicky! 

Thanks so much for your help (and for such a useful script to begin with!)!
